packages/pigeon/lib/src/kotlin/kotlin_generator.dart

```diff
@@ -654,6 +654,14 @@ if (wrapped == null) {
         indent.addScoped('{', '}', () {
           indent.writeln(
               r'val separatedMessageChannelSuffix = if (messageChannelSuffix.isNotEmpty()) ".$messageChannelSuffix" else ""');
+          String? serialBackgroundQueue;
+          if (api.methods.any((Method m) =>
+              m.taskQueueType == TaskQueueType.serialBackgroundThread)) {
+            serialBackgroundQueue = 'taskQueue';
+            serialBackgroundQueue = 'taskQueue';
```

Comment:
I think one time is probably good enough

Reply:
Fixed.